File: src/lib/config/nodes/status.ts
import type { NodeStatus } from "$lib/types/nodes";

export const NODE_STATUS_CONFIG = {
  Healthy: {
    display: 'Healthy',
    color: 'text-green-400',
    bgColor: 'bg-green-900/20',
    icon: 'CheckCircle',
    description: 'All tests passing'
  },
  Degraded: {
    display: 'Degraded',
    color: 'text-yellow-400',
    bgColor: 'bg-yellow-900/20',
    icon: 'AlertTriangle',
    description: 'Some non-critical tests failing'
  },
  Failed: {
    display: 'Failed',
    color: 'text-red-400',
    bgColor: 'bg-red-900/20',
    icon: 'XCircle',
    description: 'Critical tests failing'
  },
  Unknown: {
    display: 'Unknown',
    color: 'text-gray-400',
    bgColor: 'bg-gray-900/20',
    icon: 'HelpCircle',
    description: 'No recent test data'
  }
} as const;

export const getNodeStatusDisplayName = (s: NodeStatus) => NODE_STATUS_CONFIG[s]?.display || NODE_STATUS_CONFIG['Unknown'].display;
export const getNodeStatusColor = (s: NodeStatus) => NODE_STATUS_CONFIG[s]?.color || NODE_STATUS_CONFIG['Unknown'].color;

File: src/components/nodes/types/status.rs
use serde::{Deserialize, Serialize};
use strum_macros::EnumIter;

use crate::shared::types::metadata::TypeMetadataProvider;

#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, EnumIter)]
pub enum NodeStatus {
    Healthy,
    Degraded,
    Failed,
    Unknown,
}

impl TypeMetadataProvider for NodeStatus {
    fn id(&self) -> String { format!("{:?}", self) }
    
    fn display_name(&self) -> &str {
        match self {
            NodeStatus::Healthy => "Healthy",
            NodeStatus::Degraded => "Degraded",
            NodeStatus::Failed => "Failed",
            NodeStatus::Unknown => "Unknown",
        }
    }
    
    fn description(&self) -> &str {
        match self {
            NodeStatus::Healthy => "All tests passing",
            NodeStatus::Degraded => "Some non-critical tests failing",
            NodeStatus::Failed => "Critical tests failing",
            NodeStatus::Unknown => "No recent test data",
        }
    }
    
    fn category(&self) -> &str { "Node Health" }
    
    fn icon(&self) -> &str {
        match self {
            NodeStatus::Healthy => "CheckCircle",
            NodeStatus::Degraded => "AlertTriangle",
            NodeStatus::Failed => "XCircle",
            NodeStatus::Unknown => "HelpCircle",
        }
    }
    
    fn color(&self) -> &str {
        match self {
            NodeStatus::Healthy => "green",
            NodeStatus::Degraded => "yellow",
            NodeStatus::Failed => "red",
            NodeStatus::Unknown => "gray",
        }
    }
    
    fn metadata(&self) -> serde_json::Value {
        serde_json::json!({})
    }
}